packages/sdk/src/dom.ts

```diff
@@ -23,15 +22,22 @@ export function isStaticResourcesEqual(url1: string, url2: string): boolean {
   return relativeUrl1 === relativeUrl2;
 }
 
+export type CreateScriptHookReturn =
+  | HTMLScriptElement
+  | { script?: HTMLScriptElement; timeout?: number }
+  | void;
+
 export function createScript(
```

Comment:
I think we could add a unit test here

Reply:
I was struggling to find how to create a unit test in this case because timeout immediately throws error in simulation when remote is offline. 

Do you have an idea of how to unit test this case? 
Only on windows does it have the timeout problem so in browser its hard to reproduce

Reply:
@zhoushaw tests added, i found dom.test.ts was not running at all so i have written a handful of new tests